Sources/Kitura/RouterRequest.swift

```diff
@@ -172,19 +173,27 @@ public class RouterRequest {
         return try serverRequest.read(into: &data)
     }
 
-    /// Read the body of the request as a Codable object.
-    ///
-    /// - Parameter type: Codable object to which the body of the request will be converted.
+    /// Read the body of the request as a Codable object. It can decode JSON or URLEncoded forms.
+    /// It chooses the decoder by looking up the Content-Type header.
+    /// If there is no header it defaults to JSONDecoder.
+    /// - Parameter as: Codable object to which the body of the request will be converted.
     /// - Throws: Socket.Error if an error occurred while reading from a socket.
     /// - Throws: `DecodingError.dataCorrupted` if values requested from the payload are corrupted, or if the given data is not valid JSON.
     /// - Throws: An error if any value throws an error during decoding.
     /// - Returns: The instantiated Codable object
     public func read<T: Decodable>(as type: T.Type) throws -> T {
+        if CodableHelpers.isContentTypeURLEncoded(self) {
```

Comment:
If the content type was stored on Request, we could simply switch on it.  (see my comment above)